### ISSUE TITLE
DuckDBClient.of should support a string URL source

### ISSUE THREAD

**Issue by mbostock** [observablehq/framework#336]
You can do something similar like so, but it’s awkward:

```js
const db = DuckDBClient.of({
  gaia: {
    file: {
      name: "gaia-sample.parquet",
      url: () => "../_file/lib/gaia-sample.parquet"
    }
  }
});
```

**Comment by Fil** [comment]
The example above would not work anymore since files have a hash-based path. (But we can do `const db = DuckDBClient.of({gaia: FileAttachment("gaia-sample.parquet")});`)

Related to [this comment](https://github.com/observablehq/framework/issues/1007#issuecomment-1984331384) about supporting external urls in sql front matter.

**Comment by mbostock** [comment]
I feel like there’s a reason I was trying to demonstrate a string source rather than a `FileAttachment`, but I don’t remember now. You should also be able to use `import.meta.resolve` when #1008 is fixed.

**Comment by mbostock** [comment]
Implemented in #1098.
